### ISSUE TITLE
ID field not shown in BREAD with Belongs To Many relationship

### ISSUE THREAD

**Issue by SturmB** [thedevdojo/voyager#3456]
## Version information
 - Laravel: v5.6.29
 - Voyager: v1.1.3
 - PHP: 7.2.8
 - Database: MySQL 8.0.12

## Description
The `id` field does not appear in any BREAD view if that BREAD has a `Belongs To Many` relationship. This is best seen with Voyager's own built-in `users` table & BREAD, since it has a `Belongs To Many` relationship with `roles`.

## Steps To Reproduce
1. Install Laravel, Voyager, database, etc. The usual.
2. Login to Voyager's admin backend and navigate to the BREAD list.
4. **Edit** the BREAD for `users` and tick all of the checkboxes for the `id` field, so that they will be visible in all BREAD operation views. Make sure to save changes, natch.
5. **Browse** the BREAD for `users`.
6. Be astonished that the `id` field is not visible.

## Expected behavior
I expect the `id` field (or _any_ field, for that matter) to be visible whenever it is set as such in the table's BREAD.

## Screenshots
Easy. Here's a shot of the `users` BREAD after I have enabled all checkboxes:
<img width="1251" alt="screen shot 2018-08-06 at 12 41 54 pm" src="https://user-images.githubusercontent.com/6216224/43839924-5a7a5f40-9aed-11e8-8ce2-0f876825ee72.png">

And here's the result of browsing that very same BREAD after saving those changes:
<img width="1248" alt="screen shot 2018-08-06 at 12 42 30 pm" src="https://user-images.githubusercontent.com/6216224/43839957-6da58a4a-9aed-11e8-827d-3c72e2eac755.png">

## Additional context
This may be related to lines [57](https://github.com/the-control-group/voyager/blob/99a70e9eb65739e2d1f910d7fa5dc0a22383df45/src/Http/Controllers/VoyagerBaseController.php#L57), [142](https://github.com/the-control-group/voyager/blob/99a70e9eb65739e2d1f910d7fa5dc0a22383df45/src/Http/Controllers/VoyagerBaseController.php#L142), [189](https://github.com/the-control-group/voyager/blob/99a70e9eb65739e2d1f910d7fa5dc0a22383df45/src/Http/Controllers/VoyagerBaseController.php#L189), and [274](https://github.com/the-control-group/voyager/blob/99a70e9eb65739e2d1f910d7fa5dc0a22383df45/src/Http/Controllers/VoyagerBaseController.php#L274) in [voyager/src/Http/Controllers/VoyagerBaseController.php](voyager/src/Http/Controllers/VoyagerBaseController.php). Commenting-out those lines is not advisable, as it does cause foreign key fields to be visible (they shouldn't be) when a BREAD relationship is defined for them. I discovered this when I tried making PR #3451.

**Comment by MrCrayon** [comment]
The problem is in saving belongsToMany relationship and surfaces here:
https://github.com/the-control-group/voyager/blob/717b31f0368cb3e0cb5468fbca34ca410f8ce0b2/src/Http/Controllers/Traits/BreadRelationshipParser.php#L19

When saving belongsToMany relationship there is no column referenced in the same table as there is for belongsTo relationship, so _column_ in details field should be saved empty or not saved at all, instead a default _id_ is saved making it appear as if id is connected to a relationship field in the same BREAD.

As example in data_row defining user_belongstomany_role_relationship for users, if you change in database column value to"" you'll see id column while browsing users.

```json
{
    "model":"TCG\\Voyager\\Models\\Role",  
    "table":"roles","type":"belongsToMany",
    "column":"id",
    "key":"id",
    "label":"display_name",
    "pivot_table":"user_roles",
    "pivot":"1",
    "taggable":"0"
}
```

I'm not sure if adding an empty value in the select or removing completely that field would solve the problem without any side effect.

**Comment by MrCrayon** [comment]
AFAIK the only relationship that has a field that should be hidden in the same table is belongsTo so I think this should solve the problem:

```diff
                 $options = json_decode($row->details);
+                if ($options->type == 'belongsTo') {
                      $relationshipField = @$options->column;
                      $keyInCollection = key($dataType->{$bread_type.'Rows'}->where('field', '=', $relationshipField)->toArray());
                      array_push($forget_keys, $keyInCollection);
+                }
```

**Comment by SturmB** [comment]
That looks good, @MrCrayon. Although I'm having a little bit of trouble understanding the underlying cause as you've laid it out in your first comment (since I'm still such a n00b), I've given your proposed solution a test spin and it does, indeed, seem to fix this issue. If you don't mind, I'll go ahead and see if I can update my PR with your solution (giving you attribution for it, natch).

**Comment by MrCrayon** [comment]
@SturmB  sure go ahead :wink:

**Comment by SturmB** [comment]
This should now be fixed in #3451.
